engine/chain/chain_service.go

```diff
@@ -1,18 +1,22 @@
 package chain
 
 import (
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/policy"
 	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/query"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
 	"github.com/go-resty/resty/v2"
 	"github.com/rs/zerolog"
 )
 
 type chainService struct {
 	QueryService
+	PolicyService
 }
 
 // NewChainService creates a new chain service.
-func NewChainService(logger zerolog.Logger, httpClient *resty.Client, arcURL, arcToken, deploymentID string) Service {
+func NewChainService(logger zerolog.Logger, httpClient *resty.Client, arcCfg chainmodels.ARCConfig) Service {
 	return &chainService{
-		query.NewQueryService(logger.With().Str("chain", "query").Logger(), httpClient, arcURL, arcToken, deploymentID),
+		query.NewQueryService(logger.With().Str("chain", "query").Logger(), httpClient, arcCfg),
```

Comment:
maybe we should validate ArcConfig at this point. At least that the URL is not empty and is valid URL? And panic if it isn't?